llvm/lib/CodeGen/MachineSSAUpdater.cpp

```diff
@@ -158,9 +153,9 @@ Register MachineSSAUpdater::GetValueInMiddleOfBlock(MachineBasicBlock *BB,
     if (ExistingValueOnly)
       return Register();
     // Insert an implicit_def to represent an undef value.
-    MachineInstr *NewDef = InsertNewDef(TargetOpcode::IMPLICIT_DEF,
-                                        BB, BB->getFirstTerminator(),
-                                        VRC, MRI, TII);
+    MachineInstr *NewDef =
+        InsertNewDef(TargetOpcode::IMPLICIT_DEF, BB, BB->getFirstTerminator(),
```

Comment:
Should really use G_IMPLICIT_DEF if it's a generic virtual register

Reply:
Lane mask registers created in this pass are not generic, they have some sgpr register class and must also have S1 LLT. This is requirement for machine uniformity info to work correctly correctly.